### ISSUE TITLE
Stocks Aggregates crashes with "GE" stock

### ISSUE THREAD

**Issue by YA9** [polygon-io/client-js#178]
I'm using the Polygon Rest client.

`const client = restClient(POLYGON_API_KEY, "https://api.polygon.io");`

`client.stocks.aggregates(ticker, multiplier, timespan, currentFromDate, tomorrow.getTime(), options)`

ticker: "GE"
multiplier: 5
timespan: "minute"
currentFromDate: 1694433600000
tomorrow.getTime(): 1694548146502
options: {"limit": 5000}

This system works for the vast majority of stocks, but there are a select few, like GE, that just make my app crash with this function. It doesn't even seem like the .then or .catch run, because anytime I try to log anything, it doesn't show in my console. I've isolated the issue to this function.

Thank you for your time.

**Comment by justinpolygon** [comment]
Hey @YA9,

If you're using the latest version of the library v7.3.0 we added debugging. Mind trying to turn request tracing on and this will give you the actual url being called. Then we could see what the API is returning. Here's an example:

```javascript
import('@polygon.io/client-js').then(({ restClient }) => {
	const globalFetchOptions = {
		trace: true,
	};
	const rest = restClient("XXXX", "https://api.polygon.io", globalFetchOptions);
	rest.stocks.aggregates("GE", 1, "minute", "2023-08-01", "2023-08-01", { limit: 50000 }).then((data) => {
	    const resultCount = data.length;
	    console.log("Result count:", resultCount);
	}).catch(e => {
		console.error('An error happened:', e);
	});
});
```

I'm wondering what the API is returning and this will hopefully tell us in that you can run that same query in your browser. Mind pasting the url here and I'll debug too.

**Comment by YA9** [comment]
I'm currently using version 7.2.3. When I updated to 7.3.0, everything started going haywire. I'm not sure if the methods changed or something, but my current code is only working on 7.2.3. When I tried to enable tracing, it wasn't working on the old one.

**Comment by YA9** [comment]
Alright, I finally got it! It took me some time to fix the bugs because the results in 7.3.0 were structured differently from 7.2.3.

` LOG  Request URL:  https://api.polygon.io/v3/reference/tickers/GE
 LOG  Request Headers:  {"Authorization": "Bearer REDACTED"}
 LOG  Request URL:  https://api.polygon.io/v2/aggs/ticker/GE/range/5/minute/1694433600000/1694553981076?limit=5000
 LOG  Request Headers:  {"Authorization": "Bearer REDACTED"}
 LOG  Response Headers:  {"map": {"content-type": "application/json", "date": "Mon, 11 Sep 2023 21:26:21 GMT", "server": "nginx/1.19.2", "strict-transport-security": "max-age=15724800; includeSubDomains", "vary": "Accept-Encoding", "x-request-id": "0598b20448d8c4538f2636621c968131"}}`

**Comment by YA9** [comment]
Unfortunately, I don't get any error messages. Logs don't even work after this function runs. It's only with the GE stock too, everything else works fine. I got a stacktrace from my app as a text file. It seems to be a null pointer error. I uploaded the file for you to see here.
[stacktrace.txt](https://github.com/polygon-io/client-js/files/12600777/stacktrace.txt)

**Comment by YA9** [comment]
I just found out something interesting and potentially useful. When I run it on IOS, it works fine. On android it crashes. IOS and Android run different versions of JS. It could be an issue with the library related to JS interaction.

**Comment by justinpolygon** [comment]
Do you mind adding some logic to see if `results` is returned? What I mean, is that if I use timestamps that are in the future you'll get back something that looks like this but there is not `results` object in the json.

```
{"ticker":"GE","queryCount":0,"resultsCount":0,"adjusted":true,"status":"OK","request_id":"843e8b181c1166eea7f78de4cb885f54"}
```

But, if there are results, say for a current time then you'll get something like this:

```
{"ticker":"GE","queryCount":5,"resultsCount":3,"adjusted":true,"results":[{"v":20118,"vw":112.25,"o":112.25,"c":112.25,"h":112.25,"l":112.25,"t":1694435100000,"n":11},{"v":25164,"vw":112.268,"o":112.27,"c":111.95,"h":112.27,"l":111.95,"t":1694435400000,"n":4},{"v":114,"vw":112.1,"o":112.1,"c":112.1,"h":112.1,"l":112.1,"t":1694435700000,"n":1}],"status":"OK","request_id":"2e21936a651fdce4accb11f3ba405d29","count":3,"next_url":"https://api.polygon.io/v2/aggs/ticker/GE/range/5/minute/1694436000000/1694553981076?cursor=bGltaXQ9NSZzb3J0PWFzYw"}
```

So, there could be a null pointer to the `results` object if it's not there. Maybe some add error handleing to see if it exists first before doing the rest of your logic.

**Comment by YA9** [comment]
When I use timestamps that are in the future, I get this:
Error: Your plan doesn't include this data timeframe. Please upgrade your plan at https://polygon.io/pricing]
After I get this, the app crashes.

I cannot log anything in the function. anything in the .then() or .catch() clauses is not running because as soon as the function runs the app crashes. I have logs there, but they dont show in the terminal.

Random note: when I upgraded to version 7.3.0, the results syntax changed. Rather than returning the syntax you mentioned:
`{"ticker":"GE","queryCount":5,"resultsCount":3,"adjusted":true,"results":[{"v":20118,"vw":112.25,"o":112.25,"c":112.25,"h":112.25,"l":112.25,"t":1694435100000,"n":11},{"v":25164,"vw":112.268,"o":112.27,"c":111.95,"h":112.27,"l":111.95,"t":1694435400000,"n":4},{"v":114,"vw":112.1,"o":112.1,"c":112.1,"h":112.1,"l":112.1,"t":1694435700000,"n":1}],"status":"OK","request_id":"2e21936a651fdce4accb11f3ba405d29","count":3,"next_url":"https://api.polygon.io/v2/aggs/ticker/GE/range/5/minute/1694436000000/1694553981076?cursor=bGltaXQ9NSZzb3J0PWFzYw"}`
It's just returning the results object, without the metadata anymore. This block above is from your message btw.

**Comment by justinpolygon** [comment]
Do you mind pasting the code that you're using. Or, just the function that's making the call and here it likely crashes. Have you been able to reproduce this on your desktop/laptop? I also interested in the use case where you're trying to get aggs in the future. Are you just trying to get the latest open, high, low, close?

**Comment by YA9** [comment]
`const { restClient } = require("@polygon.io/client-js");

const client = restClient(POLYGON_API_KEY, "https://api.polygon.io", globalFetchOptions);

export function getTickerBars({ ticker, fromDate, multiplier = 1, timespan, limit = 5000, fallbackLimit = 7 }) {
    const options = {
        limit,
    };

    return new Promise((resolve, reject) => {
        const retrieveBars = (currentFromDate, fallbackCount) => {
            const currentDateString = new Date(currentFromDate).toISOString().split("T")[0];

            const today = new Date();
            const tomorrow = new Date(today);
            tomorrow.setDate(today.getDate() + 1);


            client.stocks
                .aggregates(ticker, multiplier, timespan, currentFromDate, tomorrow.getTime(), options)
                .then((data) => {
                    // console.log(data)
                    const dataLength = data.length;
                    if (data && dataLength) {
                        resolve(data);
                    } else {
                        // No data found, check fallback count
                        if (fallbackCount < fallbackLimit) {
                            // Calculate previous day and make recursive call
                            const previousFromDate = new Date(currentFromDate);
                            previousFromDate.setDate(previousFromDate.getDate() - 1);
                            retrieveBars(previousFromDate.getTime(), fallbackCount + 1);
                        } else {
                            // Reached fallback limit without data, reject the promise or handle accordingly
                            reject(new Error(`Unable to retrieve data for ${fallbackLimit} consecutive days.`));
                        }
                    }
                })

                .catch((e) => {
                    // console.log(e, ticker, multiplier, timespan, currentFromDate, tomorrow.getTime(), options);
                    reject(e);
                });
        };
        retrieveBars(fromDate, 0);
    });
}`

This is what I have. I want to fetch 1D data, and this function allows me to do that, even during weekends and holidays. It would be helpful if this feature was integrated directly into the Polygon API. Anyways, as you can see, I'm trying to log directly after calling the aggregates function.

Note: I managed to find a workaround for this issue. I switched to the "Hermes" Javascript engine from "JSC" and it works now. This is located in app.json. After I made this change, the app no longer crashes when I open GE. I get the data and everything works fine.

**Comment by timetraveler328** [comment]
@YA9 Thanks for flagging the issue about the response structure change in the latest version, I pushed patch v7.3.1 to fix it, which means that if you update versions, you'll need to look at `data.results` in the code above to get the array of results instead of it being at the top level. 

I pulled your code as well and it was working as expected with the parameters you provided, but I'm on IOS. I've not heard of an issue like this before, but we'll do some digging to see if we can figure it out. In the meantime, I wanted to warn you about the response structure bug fix in case it causes an issue for you.
